### PR TITLE
test(warehouse): verify workspace ID in stats

### PR DESCRIPTION
**Pull request by achettyiitr**
# Description

workspace ID needs to be populated in warehouse stats. Also, it should be populated in wh_staging_files and wh_uploads as well.

## Notion Ticket

https://www.notion.so/rudderstacks/Stats-testing-for-workspace-using-prometheus-926ef3734a214785827cac25f59706ac

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: warehouse/docker-compose.test.yml [hunk 1]
```diff
@@ -161,6 +161,21 @@ services:
       test: wget --no-verbose --tries=1 --spider http://localhost:8123/ping || exit 1
       interval: 1s
       retries: 25
+  wh-statsd-exporter:
```
Comment (lvrach) [inline warehouse/docker-compose.test.yml:164]:
```suggestion
  statsd-exporter:
```

Comment (achettyiitr) [reply warehouse/docker-compose.test.yml]: Done.

**Review by lvrach — APPROVED** [review]

File: warehouse/docker-compose.test.yml [hunk 2]
```diff
@@ -161,6 +161,21 @@ services:
...
+      - "8125"
+      - "8125/udp"
+    command:
+      - --statsd.listen-udp=:8125
+      - --statsd.listen-tcp=:8125
+      - --web.listen-address=:9102
+    healthcheck:
+      test: wget --no-verbose --tries=1 --spider http://localhost:9102/metrics || exit 1
+      interval: 1s
+      retries: 25
```
[the remark this answers is not in the document]
Comment (achettyiitr) [reply warehouse/docker-compose.test.yml:178]: Thanks @lvrach for pointing this out.

File: warehouse/deltalake/deltalake_test.go [hunk 3]
```diff
@@ -34,6 +34,12 @@ type TestHandle struct {
 
 var handle *TestHandle
 
+var statsToVerify = []string{
+	"warehouse_deltalake_grpcExecTime",
+	"warehouse_deltalake_grpcExecTime",
```
Comment (Jayachand) [inline warehouse/deltalake/deltalake_test.go:39]: duplicate stat

File: warehouse/bigquery/bigquery_test.go [hunk 4]
```diff
@@ -107,6 +107,8 @@ func TestBigQueryIntegration(t *testing.T) {
 		testhelper.VerifyEventsInLoadFiles(t, warehouseTest, loadFilesEventsMap())
 		testhelper.VerifyEventsInTableUploads(t, warehouseTest, tableUploadsEventsMap())
 		testhelper.VerifyEventsInWareHouse(t, warehouseTest, mergeEventsMap())
+
+		testhelper.VerifyingWorkspaceIDInStats(t)
```
Comment (Jayachand) [inline warehouse/bigquery/bigquery_test.go:111]: Name suggest: `VerifyWorkspaceIDInStats`

**Review by Jayachand — APPROVED** [review]

File: warehouse/bigquery/bigquery_test.go (hunk 4 above)
Comment (achettyiitr) [reply warehouse/bigquery/bigquery_test.go:111]: Done.

File: warehouse/deltalake/deltalake_test.go (hunk 3 above)
Comment (achettyiitr) [reply warehouse/deltalake/deltalake_test.go:39]: Done.

**Pull request merged**